File: set3.py
s = input("Enter expression: ")

# 1. Replace 'id' with single symbol
s = s.replace("id", "x")

# 2. Check valid characters
for ch in s:
    if ch not in ['x', '+', '*', '(', ')']:
        print("Rejected")
        exit()

# 3. Check parentheses balance
count = 0
for ch in s:
    if ch == '(':
        count += 1
    elif ch == ')':
        count -= 1
    if count < 0:
        print("Rejected")
        exit()

if count != 0:
    print("Rejected")
    exit()

# 4. Invalid patterns
invalid = ["++", "**", "+*", "*+", "(+", "(*", "+)", "*)"]

for p in invalid:
    if p in s:
        print("Rejected")
        exit()

# 5. Cannot start/end with operator
if s[0] in ['+', '*'] or s[-1] in ['+', '*']:
    print("Rejected")
    exit()

print("Accepted")

# 3rd from right is 'a'
s = input().strip()
print("Accepted" if len(s)>=3 and s[-3]=='a' else "Rejected")

File: set3.cpp
// SET-3
#include <iostream>
using namespace std;

string input;
int i = 0;

void E(); void Eprime(); void T(); void Tprime(); void F();

void E() { T(); Eprime(); }

void Eprime() {
    if (input[i] == '+') {
        i++; T(); Eprime();
    }
}

void T() { F(); Tprime(); }

void Tprime() {
    if (input[i] == '*') {
        i++; F(); Tprime();
    }
}

void F() {
    if (input[i] == '(') {
        i++; E();
        if (input[i] == ')') i++;
        else cout << "Error\n";
    }
    else if (input[i] == 'i' && input[i+1] == 'd') i += 2;
    else cout << "Error\n";
}

int main() {
    cin >> input;
    E();

    if (input[i] == '\0') cout << "Accepted\n";
    else cout << "Rejected\n";

    string s; cin >> s;
    if (s.length() >= 3 && s[s.length()-3] == 'a')
        cout << "Accepted\n";
    else cout << "Rejected\n";
}
